show/main.py

```diff
@@ -768,23 +771,241 @@ def get_bgp_peer():
+                        str_2_print += " [" + str(info[i]['distance']) + "/" + str(info[i]['metric']) + "]"
+                    str_length = len(str_2_print)
+                else:
+                    str_2_print += " "*(str_length - 3)
+                if "ip" in info[i]['nexthops'][j]:
+                    str_2_print += " via " + info[i]['nexthops'][j]['ip'] + ","
+                    if "interfaceName" in info[i]['nexthops'][j]:
+                        str_2_print += " " + info[i]['nexthops'][j]['interfaceName'] + ","
+                elif "directlyConnected" in info[i]['nexthops'][j]:
+                    str_2_print += " is directly connected,"
+                    if "interfaceName" in info[i]['nexthops'][j]:
+                        str_2_print += " " + info[i]['nexthops'][j]['interfaceName'] + ","
+                elif "unreachable" in info[i]['nexthops'][j]:
+                    if "reject" in info[i]['nexthops'][j]:
+                        str_2_print += " (ICMP unreachable)"
+                    elif "admin-prohibited" in info[i]['nexthops'][j]:
+                        str_2_print += " (ICMP admin-prohibited)"
+                    elif "blackhole" in info[i]['nexthops'][j]:
+                        str_2_print += " (blackhole)"
+                if "vrf" in info[i]['nexthops'][j]:
+                    str_2_print += "(vrf " + info[i]['nexthops'][j]['vrf'] + ", " + info[i]['nexthops'][j]['interfaceName'] + ","
+                if "active" not in info[i]['nexthops'][j]:
+                    str_2_print += " inactive"
+                if "onLink" in info[i]['nexthops'][j]:
+                    str_2_print += " onlink"
+                if "recursive" in info[i]['nexthops'][j]:
+                    str_2_print += " (recursive)"
+                if "source" in info[i]['nexthops'][j]:
+                    str_2_print += ", src " + info[i]['nexthops'][j]['source']
+                if "labels" in info[i]['nexthops'][j]:
+                    # MPLS labels are stored as an array (list) in json if present. Need to print through each one in list
+                    str_2_print += ", label "
+                    for k in range(0, len(info[i]['nexthops'][j]['labels'])):
+                        # MPLS labels that has value 15 or lower has special interpretation
+                        if info[i]['nexthops'][j]['labels'][k] > 15:
+                            label_string = str(info[i]['nexthops'][j]['labels'][k])
+                        else:
+                            label_string = mpls_label_code[info[i]['nexthops'][j]['labels'][k]]
+                        if k == 0:
+                            str_2_print += label_string
+                        else:
+                            str_2_print += "/" + label_string
+                # add uptime at the end of the string
+                str_2_print += " " + info[i]['uptime']
+                # print out this string
+                print str_2_print
+
+def print_show_ip_route_hdr():
```

Comment:
this can also be moved into route.py

Reply:
I see what you are saying now.. basically you want me to move all the helper routines to route.py.
To minimize the impact for now I will do this refactoring on next phase of the changes to move all helper routines to route.py
for this PR, let me keep it here in main.py for now.  I will have to work on the mock test cases next which will do this refactoring at that time as well.